File: java/src/main/java/br/com/acqio/Operator.java
package br.com.acqio;

import br.com.acqio.dto.StatusResponse;
import br.com.acqio.dto.TransactionRequest;
import br.com.acqio.dto.TransactionResponse;
import com.google.gson.Gson;
import lombok.NoArgsConstructor;
import okhttp3.MediaType;
import okhttp3.Request;
import okhttp3.RequestBody;

import java.io.IOException;
import java.util.logging.Logger;

@NoArgsConstructor
public class Operator {

    private static final Logger LOGGER = Logger.getLogger(Operator.class.getName());
    private static final String BASE_URL = "http://localhost:12030/api/transaction";

    public StatusResponse start(TransactionRequest transactionRequest) {
        Gson gson = new Gson();
        String payload = gson.toJson(transactionRequest);

        Request.Builder builder = new Request.Builder();
        builder.url(BASE_URL + "/start");
        builder.post(RequestBody.create(MediaType.parse("application/json"), payload));
        Request request = builder.build();

        try (okhttp3.Response response = HTTPConnect.getOkHttpClient().newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                String json = response.body().string();
                Debugger.log(json);
                return gson.fromJson(json, StatusResponse.class);
            }
        } catch (IOException e) {
            LOGGER.warning(e.getMessage());
        }
        return null;
    }

    public StatusResponse getStatus() {
        Gson gson = new Gson();

        Request.Builder builder = new Request.Builder();
        builder.url(BASE_URL + "/status");
        builder.get();
        Request request = builder.build();

        try (okhttp3.Response response = HTTPConnect.getOkHttpClient().newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                String json = response.body().string();
                Debugger.log(json);
                return gson.fromJson(json, StatusResponse.class);
            }
        } catch (IOException e) {
            LOGGER.warning(e.getMessage());
        }
        return null;
    }

    public StatusResponse sendCharge(int installment) {
        Gson gson = new Gson();

        Request.Builder builder = new Request.Builder();
        builder.url(BASE_URL + "/charge/" + installment);
        builder.get();
        Request request = builder.build();

        try (okhttp3.Response response = HTTPConnect.getOkHttpClient().newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                String json = response.body().string();
                Debugger.log(json);
                return gson.fromJson(json, StatusResponse.class);
            }
        } catch (IOException e) {
            LOGGER.warning(e.getMessage());
        }
        return null;
    }

    public StatusResponse revert(int id) {
        Gson gson = new Gson();

        Request.Builder builder = new Request.Builder();
        builder.url(BASE_URL + "/revert/" + id);
        builder.get();
        Request request = builder.build();

        try (okhttp3.Response response = HTTPConnect.getOkHttpClient().newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                String json = response.body().string();
                Debugger.log(json);
                return gson.fromJson(json, StatusResponse.class);
            }
        } catch (IOException e) {
            LOGGER.warning(e.getMessage());
        }
        return null;
    }

    public TransactionResponse getReceipt() {
        Gson gson = new Gson();

        Request.Builder builder = new Request.Builder();
        builder.url(BASE_URL + "/receipt/");
        builder.get();
        Request request = builder.build();

        try (okhttp3.Response response = HTTPConnect.getOkHttpClient().newCall(request).execute()) {
            if (response.isSuccessful() && response.body() != null) {
                String json = response.body().string();
                Debugger.log(json);
                return gson.fromJson(json, TransactionResponse.class);
            }
        } catch (IOException e) {
            LOGGER.warning(e.getMessage());
        }
        return null;
    }

}

File: csharp/csharp/Operator.cs
using csharp.dto;
using Newtonsoft.Json;
using System;
using System.IO;
using System.Net;

namespace csharp
{
    enum RequestType
    {
        GET, POST
    }

    class Operator
    {
        private const string BASE_URL = "http://localhost:12030/api/transaction";

        public StatusResponse Start(TransactionRequest transactionRequest)
        {          
            string url = BASE_URL + "/start";

            string payload = JsonConvert.SerializeObject(transactionRequest, Formatting.Indented);

            HttpWebRequest request = MakeRequest(url, RequestType.POST);

            using (StreamWriter swJsonPayload = new StreamWriter(request.GetRequestStream()))
            {
                swJsonPayload.Write(payload);
                swJsonPayload.Close();
            }

            string response = SendRequest(request);

            if (response != null)
            {
                return JsonConvert.DeserializeObject<StatusResponse>(response);
            }

            return null;
        }

        public StatusResponse GetStatus()
        {

            string url = BASE_URL + "/status";

            HttpWebRequest request = MakeRequest(url, RequestType.GET);

            string response = SendRequest(request);

            if (response != null)
            {
                return JsonConvert.DeserializeObject<StatusResponse>(response);
            }

            return null;
        }

        public StatusResponse SendCharge(int installment)
        {
            string url = BASE_URL + "/charge/" + installment;

            HttpWebRequest request = MakeRequest(url, RequestType.GET);

            string response = SendRequest(request);

            if (response != null)
            {
                return JsonConvert.DeserializeObject<StatusResponse>(response);
            }

            return null;
        }

        public StatusResponse Revert(int id)
        {
            string url = BASE_URL + "/revert/" + id;

            HttpWebRequest request = MakeRequest(url, RequestType.GET);

            string response = SendRequest(request);

            if (response != null)
            {
                return JsonConvert.DeserializeObject<StatusResponse>(response);
            }

            return null;
        }

        public TransactionResponse GetReceipt()
        {
            string url = BASE_URL + "/receipt/";

            HttpWebRequest request = MakeRequest(url, RequestType.GET);

            string response = SendRequest(request);

            if (response != null)
            {
                return JsonConvert.DeserializeObject<TransactionResponse>(response);
            }

            return null;
        }

        private HttpWebRequest MakeRequest(string url, RequestType requestType)
        {
            HttpWebRequest request = (HttpWebRequest)HttpWebRequest.Create(url);
            request.Method = requestType.ToString();
            request.ContentType = "application/json";
            return request;
        }

        private string SendRequest(HttpWebRequest request)
        {
            try
            {
                HttpWebResponse response = (HttpWebResponse)request.GetResponse();

                using (Stream responseStream = response.GetResponseStream())
                {
                    if (responseStream != null)
                    {
                        using (StreamReader reader = new StreamReader(responseStream))
                        {
                            string responseBody = reader.ReadToEnd();
                            Console.WriteLine(responseBody);
                            return responseBody;
                        }
                    }
                }
            }
            catch (Exception e)
            {
                Debugger.Log(e);
            }

            return null;
        }

    }
}